### PR TITLE
HDFS-15100. RBF: Print stacktrace when DFSRouter fails to fetch/parse JMX output from NameNode.

### PR DESCRIPTION
**Pull request by aajisaka**
JIRA: https://issues.apache.org/jira/browse/HDFS-15100

### CODE REVIEW

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 14s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  18m 21s |  trunk passed  |
| +1 :green_heart: |  compile  |   0m 34s |  trunk passed  |
| +1 :green_heart: |  checkstyle  |   0m 24s |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 38s |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  13m 25s |  branch has no errors when building and testing our client artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 53s |  trunk passed  |
| +0 :ok: |  spotbugs  |   1m  7s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   1m  5s |  trunk passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 30s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 26s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 26s |  the patch passed  |
| -0 :warning: |  checkstyle  |   0m 17s |  hadoop-hdfs-project/hadoop-hdfs-rbf: The patch generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0)  |
| +1 :green_heart: |  mvnsite  |   0m 30s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedclient  |  13m 28s |  patch has no errors when building and testing our client artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 50s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   1m 10s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  |   9m 27s |  hadoop-hdfs-rbf in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 33s |  The patch does not generate ASF License warnings.  |
|  |   |  65m 47s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hdfs.server.federation.router.TestRouterRpc |
|   | hadoop.hdfs.server.federation.router.TestRouterClientRejectOverload |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1800/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1800 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux c4dcbbafe1d6 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / aba3f6c |
| Default Java | 1.8.0_232 |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1800/1/artifact/out/diff-checkstyle-hadoop-hdfs-project_hadoop-hdfs-rbf.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1800/1/artifact/out/patch-unit-hadoop-hdfs-project_hadoop-hdfs-rbf.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1800/1/testReport/ |
| Max. process+thread count | 2453 (vs. ulimit of 5500) |
| modules | C: hadoop-hdfs-project/hadoop-hdfs-rbf U: hadoop-hdfs-project/hadoop-hdfs-rbf |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1800/1/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

File: hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/FederationUtil.java [hunk 1]
```diff
@@ -98,13 +98,10 @@ public static JSONArray getJmx(String beanQuery, String webAddress,
       JSONObject json = new JSONObject(jmxOutput);
       ret = json.getJSONArray("beans");
     } catch (IOException e) {
-      LOG.error("Cannot read JMX bean {} from server {}: {}",
-          beanQuery, webAddress, e.getMessage());
-    } catch (JSONException e) {
```
Comment (goiri) [inline hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/FederationUtil.java]: I would keep this one separately as we shouldn't need more details on this if the JSON parsing fails.

File: hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/FederationUtil.java [hunk 2]
```diff
@@ -98,13 +98,10 @@ public static JSONArray getJmx(String beanQuery, String webAddress,
...
-      LOG.error("Cannot parse JMX output for {} from server {}: {}",
-          beanQuery, webAddress, e.getMessage());
+      LOG.error("Cannot read JMX bean {} from server {}: ",
+          beanQuery, webAddress, e);
     } catch (Exception e) {
-      LOG.error("Cannot parse JMX output for {} from server {}: {}",
```
Comment (goiri) [inline hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/FederationUtil.java]: I guess this is the one we need to capture.
Let's remove the last ":" and leave it as:
LOG.error("Cannot parse JMX output for {} from server {}",
    beanQuery, webAddress, e);

File: hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/FederationUtil.java [hunk 3]
```diff
@@ -98,13 +98,10 @@ public static JSONArray getJmx(String beanQuery, String webAddress,
       JSONObject json = new JSONObject(jmxOutput);
       ret = json.getJSONArray("beans");
     } catch (IOException e) {
-      LOG.error("Cannot read JMX bean {} from server {}: {}",
-          beanQuery, webAddress, e.getMessage());
-    } catch (JSONException e) {
-      LOG.error("Cannot parse JMX output for {} from server {}: {}",
-          beanQuery, webAddress, e.getMessage());
+      LOG.error("Cannot read JMX bean {} from server {}: ",
```
Comment (goiri) [inline hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/FederationUtil.java]: Leave as:
LOG.error("Cannot parse JMX output for {} from server {}",
beanQuery, webAddress, e);

**Comment by aajisaka** [comment]
Thanks @tasanuma and @goiri  for the reviews.
Updated the pull request.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |  24m 50s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  18m 19s |  trunk passed  |
| +1 :green_heart: |  compile  |   0m 34s |  trunk passed  |
| +1 :green_heart: |  checkstyle  |   0m 24s |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 37s |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  13m 23s |  branch has no errors when building and testing our client artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 51s |  trunk passed  |
| +0 :ok: |  spotbugs  |   1m  7s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   1m  4s |  trunk passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 29s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 27s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 27s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 17s |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   0m 29s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedclient  |  13m 14s |  patch has no errors when building and testing our client artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 48s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   1m 10s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   9m 26s |  hadoop-hdfs-rbf in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 33s |  The patch does not generate ASF License warnings.  |
|  |   |  88m 59s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1800/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1800 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux ba921b088d8b 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 782c055 |
| Default Java | 1.8.0_232 |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1800/2/testReport/ |
| Max. process+thread count | 2329 (vs. ulimit of 5500) |
| modules | C: hadoop-hdfs-project/hadoop-hdfs-rbf U: hadoop-hdfs-project/hadoop-hdfs-rbf |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1800/2/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Pull request merged**